packages/api/src/external/carequality/document/create-outbound-document-retrieval-req.ts

```diff
@@ -1,35 +1,32 @@
-import { Organization } from "@metriport/core/domain/organization";
+import { Patient } from "@metriport/core/domain/patient";
 import { capture } from "@metriport/core/util/notifications";
 import {
   OutboundDocumentQueryResp,
   OutboundDocumentRetrievalReq,
 } from "@metriport/ihe-gateway-sdk";
 import dayjs from "dayjs";
 import { chunk } from "lodash";
-import { createPurposeOfUse, isGWValid } from "../shared";
+import { createPurposeOfUse, getCqInitiator, getSystemUserName, isGWValid } from "../shared";
 import { DocumentReferenceWithMetriportId } from "./shared";
 
 const SUBJECT_ROLE_CODE = "106331006";
 const SUBJECT_ROLE_DISPLAY = "Administrative AND/OR managerial worker";
 export const maxDocRefsPerDocRetrievalRequest = 5;
 
-export function createOutboundDocumentRetrievalReqs({
+export async function createOutboundDocumentRetrievalReqs({
```

Comment:
Wouldn't it be easier/better to keep this function working data in memory, not returning a Promise, and pass the initiator to it, instead of the cxId/organization as it was before?

From my perspective, this function just gets some data and builds an object that represents the DR request, essentially a simple data transformer/converter.

By having it call `getCqInitiator()`, now it has more logic as to how to obtain some of the information needed to build the response, not just how convert that data into it.

(not to mention that tests just got more complicated with spying/mock return.

Reply:
I guess that's true..

Reply:
Now that we're passing the initiator this function doesn't need to be async anymore, so we can revert to the original return type `OutboundDocumentRetrievalReq[]`, right?